src/common/types/types.cpp

```diff
@@ -594,7 +670,11 @@ std::unique_ptr<LogicalType> LogicalType::deserialize(Deserializer& deserializer
         extraTypeInfo = StructTypeInfo::deserialize(deserializer);
     } break;
     default:
-        extraTypeInfo = nullptr;
+        if (typeID == LogicalTypeID::DECIMAL) {
```

Comment:
ditto

Reply:
This might not be possible with deserialization, because extraTypeInfo is initially nullptr.